app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [OPTIONAL] I think it would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the #update_posts_counter method every time a new post has been saved to database, then make the `update_posts_counter` method private 🙃 🙃